### ISSUE TITLE
No duplicated-code error/warning?

### ISSUE THREAD

**Issue by wulfgarpro** [PyCQA/pylint#1751]
### Steps to reproduce
1. copy code into script file
2. run pylint against said file
3. assess result

### Current behavior
No duplicated-code error/warning.

### Expected behavior
A duplicated-code error/warning.

### pylint --version output
$ pylint --version
No config file found, using default configuration
pylint 1.7.4, 
astroid 1.5.3
Python 3.6.3 (default, Oct  3 2017, 21:45:48) 
[GCC 7.2.0]

script.py
--
```python
"""
module
"""

def go1(x_arg=1):
    """
    go
    """
    if x_arg is 1:
        print(x_arg)


def go2(x_arg=1):
    """
    go
    """
    if x_arg is 1:
        print(x_arg)


def main():
    """
    main
    """
    print("main")
    go1()
    go2()

if __name__ == "__main__":
    main()
```

**Comment by PCManticore** [comment]
Unfortunately pylint can't currently find duplicated code in the same file. While technically it would be possible to do so, adding support for this will complicate the currently brittle algorithm, so I'm closing this issue.
